src/d3a/models/myco_matcher/pay_as_clear.py

```diff
@@ -15,33 +15,27 @@
 You should have received a copy of the GNU General Public License
 along with this program.  If not, see <http://www.gnu.org/licenses/>.
 """
+
+from d3a.models.myco_matcher.pay_as_bid import PayAsBidMatch
+
 import math
 from logging import getLogger
 from collections import OrderedDict
 
-from d3a.models.market.two_sided_pay_as_bid import TwoSidedPayAsBid
 from d3a.models.market.market_structures import MarketClearingState, BidOfferMatch, \
     TradeBidOfferInfo, Clearing
-from d3a_interface.constants_limits import ConstSettings, GlobalConfig
+from d3a_interface.constants_limits import ConstSettings
 from d3a.d3a_core.util import add_or_create_key
 from d3a.constants import FLOATING_POINT_TOLERANCE
 
-log = getLogger(__name__)
 
+log = getLogger(__name__)
 
-class TwoSidedPayAsClear(TwoSidedPayAsBid):
 
-    def __init__(self, time_slot=None, bc=None, notification_listener=None, readonly=False,
-                 grid_fee_type=ConstSettings.IAASettings.GRID_FEE_TYPE,
-                 grid_fees=None, name=None, in_sim_duration=True):
-        super().__init__(time_slot, bc, notification_listener, readonly,
-                         grid_fee_type, grid_fees, name,
-                         in_sim_duration=in_sim_duration)
+class PayAsClear(PayAsBidMatch):
```

Comment:
I think that a common abstract base class for the matcher would be great to have, in order to define methods that all matcher classes will need to implement, and to not need to inherit the PayAsClear from PayAsBid. That way you can put the sorting methods in the abc, and PayAsClear and PayAsBid can both inherit from the same base class. That way the class hierarchy will be flat.

Reply:
Added: https://github.com/gridsingularity/d3a/pull/1121/commits/8e7aa92b935c04751d47fe6e1b16b41579354e2f